File: epi_judge_java/epi/IntersectSortedArrays.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class IntersectSortedArrays {

    @EpiTest(testDataFile = "intersect_sorted_arrays.tsv")
    public static List<Integer> intersectTwoSortedArrays(List<Integer> a, List<Integer> b) {
        List<Integer> ans = new ArrayList<>();
        if (a.size() > b.size()) {
            var temp = a;
            a = b;
            b = temp;
        }
        for (int i = 0; i < a.size(); i++) {
            if (i > 0 && a.get(i).equals(a.get(i - 1))) continue;
            if (Collections.binarySearch(b, a.get(i)) >= 0) {
                ans.add(a.get(i));
            }
        }
        return ans;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "IntersectSortedArrays.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/intersect_sorted_arrays.cc
#include <vector>

#include "test_framework/generic_test.h"

using namespace std;

vector<int> IntersectTwoSortedArrays(const vector<int> &a, const vector<int> &b) {
    set<int> unique;
    if (a.size() > b.size()) {
        for (int num: b) {
            if (std::binary_search(a.begin(), a.end(), num)) {
                unique.insert(num);
            }
        }
    } else {
        for (int num: a) {
            if (std::binary_search(b.begin(), b.end(), num)) {
                unique.insert(num);
            }
        }
    }
    vector<int> ans(unique.begin(), unique.end());
    return ans;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"A", "B"};
    return GenericTestMain(
            args, "intersect_sorted_arrays.cc", "intersect_sorted_arrays.tsv",
            &IntersectTwoSortedArrays, DefaultComparator{}, param_names);
}
